src/NUnitFramework/framework/Internal/TestParameters.cs

```diff
@@ -30,52 +32,55 @@ namespace NUnit.Framework.Internal
     /// TestParameters is the abstract base class for all classes
     /// that know how to provide data for constructing a test.
     /// </summary>
-    public abstract class TestParameters : ITestData, IApplyToTest
+    public abstract class TestParameters : ITestData, IApplyToTest, IAlignArguments
     {
+        #region Constants
+        private static readonly Array ArrayUndefined = new object[0];
+
+        internal const string ExceptionParameterCount = "Method expects {0} arguments but test case supplied {1}.";
+        #endregion
+
         #region Constructors
 
         /// <summary>
-        /// Default Constructor creates an empty parameter set
+        /// Default constructor inititalizes default state.
         /// </summary>
-        public TestParameters()
+        protected TestParameters()
         {
             RunState = RunState.Runnable;
             Properties = new PropertyBag();
         }
 
         /// <summary>
-        /// Construct a parameter set with a list of arguments
+        /// Construct a parameter set with a list of <paramref name="args">arguments</paramref>.
```

Comment:
Keep in mind that VS intellisense won't show the contents of `<paramref>` and `<see>` elements. Personally I wouldn't bother and would make it self-closing, but if you do include contents make sure that it makes sense when read either way like you did here.